web/javascript/formatter.js

```diff
@@ -6,287 +6,304 @@
-                    explanation = "Downloading from this peer";
-                    break;
-                case "d":
-                    explanation = "We would download from this peer if they'd let us";
-                    break;
-                case "U":
-                    explanation = "Uploading to peer";
-                    break;
-                case "u":
-                    explanation = "We would upload to this peer if they'd ask";
-                    break;
-                case "K":
-                    explanation = "Peer has unchoked us, but we're not interested";
-                    break;
-                case "?":
-                    explanation = "We unchoked this peer, but they're not interested";
-                    break;
-                case "E":
-                    explanation = "Encrypted Connection";
-                    break;
-                case "H":
-                    explanation = "Peer was discovered through Distributed Hash Table (DHT)";
-                    break;
-                case "X":
-                    explanation = "Peer was discovered through Peer Exchange (PEX)";
-                    break;
-                case "I":
-                    explanation = "Peer is an incoming connection";
-                    break;
-                case "T":
-                    explanation = "Peer is connected via uTP";
-                    break;
-                };
-
-                if (!explanation) {
-                    formattedFlags.push(flag);
-                } else {
-                    formattedFlags.push("<span title=\"" + flag + ': ' + explanation + "\">" + flag + "</span>");
-                };
-            };
-
-            return formattedFlags.join('');
+  var speed_K = 1000;
+  var speed_K_str = 'kB/s';
+  var speed_M_str = 'MB/s';
+  var speed_G_str = 'GB/s';
+
+  var size_K = 1000;
+  var size_B_str = 'B';
+  var size_K_str = 'kB';
+  var size_M_str = 'MB';
+  var size_G_str = 'GB';
+  var size_T_str = 'TB';
+
+  var mem_K = 1024;
+  var mem_B_str = 'B';
+  var mem_K_str = 'KiB';
+  var mem_M_str = 'MiB';
+  var mem_G_str = 'GiB';
+  var mem_T_str = 'TiB';
+
+  return {
+    /*
+     *   Format a percentage to a string
+     */
+    percentString: function (x) {
+      if (x < 10.0) {
+        return x.toTruncFixed(2);
+      } else if (x < 100.0) {
+        return x.toTruncFixed(1);
+      } else {
+        return x.toTruncFixed(0);
+      }
+    },
+
+    /*
+     *   Format a ratio to a string
+     */
+    ratioString: function (x) {
+      if (x === -1) {
+        return 'None';
+      }
+      if (x === -2) {
+        return '&infin;';
+      }
+      return this.percentString(x);
+    },
+
+    /**
+     * Formats the a memory size into a human-readable string
+     * @param {Number} bytes the filesize in bytes
+     * @return {String} human-readable string
+     */
+    mem: function (bytes) {
+      if (bytes < mem_K) return [bytes, mem_B_str].join(' ');
```

Comment:
Putting `if` and `return` on the same line is not something old formatter did and not something I'd like to have. OTOH, new formatter doesn't add or remove braces (a pity, but not saying old one did either, have no idea) and if there're braces with only a single statement inside it doesn't squash it all into a single line, so maybe we could add a couple here.

Reply:
I've added eslint + a curly rule for this in 56350680945a86f7ec6f79a0ed6a2f28a78c4c8b